pkg/skaffold/survey/config_test.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package survey
+
+import (
+	"testing"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestSurveyPrompt(t *testing.T) {
+	tests := []struct {
+		description string
+		s           config
+		expected    string
+	}{
+		{
+			description: "hats survey",
+			s:           hats,
+			expected: `Help improve Skaffold with our 2-minute anonymous survey: run 'skaffold survey'
+`,
+		},
+		{
+			description: "not hats survey",
+			s: config{
+				id:         "foo",
+				promptText: "Looks like you are using foo feature. Help improve Skaffold foo feature and take this survey",
+				expiresAt:  time.Date(2021, time.August, 14, 00, 00, 00, 0, time.UTC),
+			},
+			expected: `Looks like you are using foo feature. Help improve Skaffold foo feature and take this survey: run 'skaffold survey -id foo'
+`,
+		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			t.CheckDeepEqual(test.s.prompt(), test.expected)
+		})
+	}
+}
+
+func TestSurveyActive(t *testing.T) {
+	tests := []struct {
+		description string
+		s           config
+		expected    bool
+	}{
+		{
+			description: "no expiry",
+			s:           hats,
+			expected:    true,
+		},
+		{
+			description: "expiry in past",
+			s: config{
+				id:        "expired",
+				expiresAt: time.Date(2020, time.August, 14, 00, 00, 00, 0, time.UTC),
+			},
+		},
+		{
+			description: "expiry in future",
+			s: config{
+				id:        "active",
+				expiresAt: time.Date(2022, time.August, 14, 00, 00, 00, 0, time.UTC),
```

Comment:
This test will break after next year since `isActive()` isn't actually using `today`.  But better IMHO to make this relative to now:
```suggestion
				expiresAt: time.Now().Add(5*time.Hour),
```

Reply:
done